mesheryctl/internal/cli/root/models/import.go

```diff
@@ -0,0 +1,266 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package models
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/system"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1beta1"
+	"github.com/layer5io/meshkit/models/meshmodel/registry"
+	extract "github.com/layer5io/meshkit/utils"
+	"github.com/layer5io/meshkit/utils/walker"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var importModelCmd = &cobra.Command{
+	Use:   "import",
+	Short: "import model",
+	Long:  "import a model from the registry",
+	Example: `
+// Import a model
+mesheryctl exp model import <URI>
+
+// Import a model from a local directory
+mesheryctl exp model import <path to model>
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return system.ErrGetCurrentContext(err)
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl exp model import <URI/path to Model>\nRun 'mesheryctl exp model import --help' to see detailed help message"
+		if len(args) == 0 {
+			if err := cmd.Usage(); err != nil {
+				return err
+			}
+			return utils.ErrInvalidArgument(errors.New(errMsg))
+		}
+		return nil
+	},
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if validURL := govalidator.IsURL(args[0]); !validURL {
+			return importFromLocalFile(args[0])
+		} else if isGithubURL := strings.Contains(args[0], "github.com"); isGithubURL {
+			return importFromGithubURL(args[0])
+		}
+		return nil
+	},
+}
+
+func importFromLocalFile(filePath string) error {
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		return utils.ErrFileRead(errors.Errorf("File path does not exist. Path provided: %s", filePath))
+	}
+
+	if strings.HasSuffix(filePath, ".zip") {
+		return processZip(filePath)
+	}
+
+	if strings.HasSuffix(filePath, ".tar.gz") || strings.HasSuffix(filePath, ".tgz") {
+		return processTarGz(filePath)
+	}
+
+	dirPath := filePath
+	// fmt.Println("Directory path:", dirPath)
+	return processDirectory(dirPath)
+}
+
+func processZip(zipFilePath string) error {
+	homeDir, _ := os.UserHomeDir()
+	tmpDir := homeDir + "/.meshery/temp"
```

Comment:
This is not the only process that uses temporary storage. 

Where is the common function for 1) finding, 2) using, and 3) cleaning up temporary storage?